### PR TITLE
Kodi v18 with InputStream Adaptive

### PR DESCRIPTION
**Pull request by sephalon**
###### Motivation for this change

My primary motivation was to get the Amazon VOD plugin with InputStream Adaptive backend up and running.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by domenkozar** [comment]
lgtm

**Comment by Infinisil** [comment]
Any update on this?

**Comment by sephalon** [comment]
Updated to Kodi v18rc5.2 and inputstream-adaptive 2.3.12; still works so far.

**Comment by cpages** [comment]
lgtm, but this update will probably break a bunch of plugins that we'll need to fix afterwards

**Comment by sephalon** [comment]
Updated to Kodi v18.0 final now. Also updated some dependencies that I previously forgot about.

inputstream.adaptive cannot be updated before https://github.com/peak3d/inputstream.adaptive/issues/230 hasn't been fixed. 2.3.12 still works though.

File: pkgs/applications/video/kodi/default.nix [hunk 1]
```diff
@@ -72,9 +69,9 @@ let
 
   ffmpeg = kodiDependency rec {
     name    = "FFmpeg";
-    version = "3.1.11";
-    rev     = "${version}-${rel}-17.5"; # TODO: change 17.5 back to ${kodiVersion}
-    sha256  = "0nc4sb6v1g3l11v9h5l9n44a8r40186rcbp2xg5c7vg6wcpjid13";
+    version = "4.0.3";
+    rev     = "${version}-${rel}-RC5";
```
Comment (peterhoeg) [inline pkgs/applications/video/kodi/default.nix]: Is this still correct?

File: pkgs/applications/video/kodi/default.nix [hunk 2]
```diff
@@ -86,29 +83,29 @@ let
...
   # Doesn't work ATM though so we just use them for the src
 
-  libdvdcss = kodiDependency {
+  libdvdcss = kodiDependency rec {
     name              = "libdvdcss";
-    version           = "20160215";
-    rev               = "2f12236bc1c92f73c21e973363f79eb300de603f";
-    sha256            = "198r0q73i55ga1dvyqq9nfcri0zq08b94hy8671lg14i3izx44dd";
+    version           = "1.4.2";
+    rev               = "${version}-${rel}-Beta-5";
```
Comment (peterhoeg) [inline pkgs/applications/video/kodi/default.nix]: And this?

File: pkgs/applications/video/kodi/default.nix (hunk 1 above)
Comment (cpages) [reply pkgs/applications/video/kodi/default.nix]: Latest from upstream, so yes

File: pkgs/applications/video/kodi/default.nix (hunk 2 above)
Comment (cpages) [reply pkgs/applications/video/kodi/default.nix]: Same here, latest from upstream

**Review by cpages — APPROVED** [review]
Works for me

File: pkgs/applications/video/kodi/default.nix (hunk 1 above)
Comment (sephalon) [reply pkgs/applications/video/kodi/default.nix]: Yeah, seems like they don't bump all their dependencies for a release…

**Comment by aanderse** [comment]
Any chance of this getting in 19.03?

**Comment by worldofpeace** [comment]
If @cpages says this is fine, I'll merge this.

However all the plugins that we have we'll need to be updated. That's not a very good change to have between releases.

**Comment by cpages** [comment]
From the wiki it says that most plugins should just work, and I see that just a few fail to build (some of which seem already broken anyway #25490). I think it's worth to have this in 19.03

**Comment by cpages** [comment]
> If @cpages says this is fine, I'll merge this.
> 
> However all the plugins that we have we'll need to be updated because of breakage.

Just a heads-up that I updated all the ones that were failing to build. I can't test all of them, though, so let's hope for the best

**Comment by aanderse** [comment]
@cpages I found an issue with `inputstream-adaptive` when I tried to load the Netflix plugin.
> Error Contents: No module named Cryptodome.Random

Looks like `python2Packages.pycryptodomex` is required.

**Comment by cpages** [comment]
@aanderse if I read correctly, pycryptodomex is a dependency of the netflix plugin itself, not of inputstream-adaptive. But then it also lists Libwidevine as a requirement, so maybe you can open a separate issue?

**Comment by aanderse** [comment]
@cpages I thought I had read somewhere it was a dependency of inputstream-adaptive itself, not kodi... but I can't find a reference to that now (and you aren't having this issue) so that's obviously wrong.
If this python library is a dependency and assumes the system already has it installed I guess the two options are to include with kodi itself, or make a nix package for netflix.

**Pull request merged**